src/disc/stk/Albany_STKDiscretization.cpp

```diff
@@ -2089,19 +2096,17 @@ STKDiscretization::computeNodeSets()
     stk::mesh::get_selected_entities(ns_selector, bulkData->buckets(NODE_RANK), nodes);
 
     // Remove nodes that are not owned according to Tpetra/Epetra
-    bool done = false;
-    while (not done) {
-      done = true;
-      auto dof_mgr = getNodeDOFManager();
-      for (auto it=nodes.begin(); it!=nodes.end(); ++it) {
-        auto gid = stk_gid(*it);
-        if (not dof_mgr->indexer()->isLocallyOwnedElement(gid)) {
-          nodes.erase(it);
-          done = false;
-          break;
-        }
+    std::vector<stk::mesh::Entity>::iterator it = nodes.begin();
+    auto dof_mgr = getNodeDOFManager();
+    while(it != nodes.end()){
+      auto gid = stk_gid(*it);
+      if (not dof_mgr->indexer()->isLocallyOwnedElement(gid)){
+        it = nodes.erase(it);
```

Comment:
Ah, nice. I was scared by the fact that erasing invalidates pointers, but I suspect the compiler does not caches nodes.end() in the while condition, so this is fine.